src/qrules/io/_dot.py

```diff
@@ -497,14 +497,15 @@ def _collapse_graphs(
         )
         for g in graphs
     }
+
     for transition in graphs:
         topology = transition.topology
         group = transition_groups[topology]
         for state_id, state in transition.states.items():
             if isinstance(state, State):
                 particle = state.particle
             else:
-                particle, _ = state
+                particle = state  # ??? convert state to actual particle? || change particle to property?
```

Comment:
As your comment suggests, this block is probably better extracted as a function that does something like

```python
def _strip_properties(state: Any) -> Any:
    ...
```

In the end, the whole function does something like:

```python
def _collapse_graphs(graphs: Iterable[Transition]) -> list[FrozenTransition[tuple, None]]: ...
```

Note that the typing of the edges and nodes has been remove. Theoretically, you can make finegrain the typing here with [`TypeVar`](https://docs.python.org/3/library/typing.html#typing.TypeVar) and [`@overload`](https://docs.python.org/3/library/typing.html#typing.overload), but as these are internal function, I think it's fine to stick with `Any` for now.